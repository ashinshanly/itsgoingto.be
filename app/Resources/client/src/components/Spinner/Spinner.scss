@import "bourbon";
@import "neat";
@import "variables";

.spinner {
  display: inline-block;
  position: relative;
  text-align: center;
  vertical-align: middle;
}

.spinner_dot {
  animation-name: pulse;
  animation-duration: 1s;
  animation-iteration-count: infinite;
  animation-timing-function: ease-in-out;
  height: 1em;
  width: 1em;
  border-radius: 50%;
  display: inline-block;
  vertical-align: top;

  background-color: #999;

  &.spinner_dot--second {
    animation-delay: 160ms;
    margin-left: 1em;
  }
  &.spinner_dot--third {
    animation-delay: 320ms;
    margin-left: 1em;
  }
}

.btn > .spinner {
  margin-right: 2em;

  &:only-child {
    margin-right: 0;
  }
}

@keyframes pulse {
  0%, 80%, 100% { opacity: 0; }
  40% { opacity: 1; }
}
