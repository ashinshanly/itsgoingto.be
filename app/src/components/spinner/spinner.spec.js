/* global expect */
import React from 'react'
import { shallow } from 'enzyme'
import Spinner from './spinner'

describe('(Component) Spinner', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Spinner />)
  })

  describe('(Render) snapshot', () => {
    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
