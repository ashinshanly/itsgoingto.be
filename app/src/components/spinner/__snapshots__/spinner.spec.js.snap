// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Spinner (Render) snapshot renders correctly 1`] = `
<div
  className="spinner"
>
  <span
    className="spinner_dot spinner_dot--first"
  />
  <span
    className="spinner_dot spinner_dot--second"
  />
  <span
    className="spinner_dot spinner_dot--third"
  />
</div>
`;
